File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
import math
from itertools import permutations

def is_prime(x):  
    for i in range(2, int(x**0.5)+1):
        if x % i == 0:
            return False
    return True

def solution(numbers):
    answer = 0
    num = list(numbers)
    num_lst = []
    for i in range(1, len(num)+1):
        num_lst += list(permutations(num, i))
    lst = []
    for i in num_lst:
        lst.append(int(''.join(i)))
    lst = list(set(lst))
    
    for i in lst:
        if i <= 1:
            continue
        elif is_prime(i):
            answer += 1
    
    return answer

# def solution(numbers):
#     answer = 0
#     n = list(numbers)

#     a = []
#     for i in range(1, len(n)+1):
#         a += list(permutations(n, i))   #경우의 수 반환

#     b = []
#     for i in a:
#         b.append(int(''.join(i)))   
#     b = list(set(b))    #경우의 수를 int형으로 담은 배열 b 선언

#     for i in b:
#         if i <= 1:
#             continue    #1 이하면 무시
#         elif is_prime(i):   # 소수면
#             answer += 1     # answer 증가

#     return answer

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    static char[] arr;
    static Set<Integer> set = new HashSet<>();
    static boolean[] visited;
    static char[] selected;
    public int solution(String numbers) {
        int answer = 0;
        arr = numbers.toCharArray();
        visited = new boolean[arr.length];
        for (int i = 1; i <= arr.length; i++) {
            selected = new char[i];
            permutation(i, 0);
        }
        for (Integer s : set) {
            if (isPrime(s)) answer++;
        }
        return answer;
    }
    
    static void permutation(int len, int cnt) {
        if (cnt == len) {
            String num = "";
            for (int i = 0; i < len; i++) {
                num += String.valueOf(selected[i]);
            }
            set.add(Integer.parseInt(num));
            return;
        }
        for (int i = 0; i < arr.length; i++) {
            if (!visited[i]) {
                visited[i] = true;
                selected[cnt] = arr[i];
                permutation(len, cnt+1);
                visited[i] = false;
            }
        }
    }
    
    static boolean isPrime(int n) {
        if (n == 0 || n == 1) return false;
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) return false;
        }
        return true;
    }
}